### PR TITLE
ref(js): Remove JQueryXHR types + jqXHR var name

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/api.tsx [hunk 1]
```diff
@@ -345,25 +370,33 @@ export class Client {
     /**
      * Called when the request completes with a 2xx status
      */
-    const successHandler = (responseData: any, textStatus: string, xhr: JQueryXHR) => {
+    const successHandler = (
+      resp: ResponseMeta,
+      textStatus: string,
+      responseData: any
```
Comment (EvanPurkhiser) [inline static/app/api.tsx:376]: I changed the parameter order here to match with errorHandler

Comment (billyvg) [reply static/app/api.tsx:376]: Is this a private function?

**Review by billyvg — APPROVED** [review]
Any getsentry changes required?

File: static/app/api.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply static/app/api.tsx:376]: Yeah, it's only defined inside of `request`

**Pull request merged**